src/java.base/share/classes/java/lang/reflect/Proxy.java

```diff
@@ -1314,6 +1314,46 @@ public MethodHandles.Lookup run() {
         });
     }
 
+    /*
+     * Invoke the default method of the given proxy with an explicit caller class.
+     *
+     * @throws IllegalAccessException if the proxy interface is inaccessible to the caller
+     *         if caller is non-null
+     */
+    static Object invokeDefault(Object proxy, Method method, Object[] args, Class<?> caller)
```

Comment:
This is the piece of code that will now unify calling methods both from `MethodHandleProxies` (where this code is copied from) and `ReflectAccess`, which was using a different way of calling methods.